sqlx/format.ts

```diff
@@ -16,11 +15,13 @@ const JS_BEAUTIFY_OPTIONS: JsBeautifyOptions = {
 
 const MAX_SQL_FORMAT_ATTEMPTS = 5;
 
-export function format(text: string, fileExtension: string) {
+export type SqlLanguage = sqlFormatter.SqlLanguage;
```

Comment:
Adding a dependenncy to `WarehouseType (df/core/adapters)` looks unwelcome, so I took the sql-formatter argument type here. But it may seem smarter to define `Record<WarehouseType, SqlLanguage>` here.

Reply:
agreed - while it is not ideal to add a dependency to `core` here, I think it's still probably cleaner to use `WarehouseType` in `format`'s API, and hide `sql-formatter` types from the caller (and so move `Record<WarehouseType, SqlLanguage>` here, as you say).

Reply:
Thanks for reviewing! I agree that moving the definitions around would look cleaner.

But adding a dependency in sqlx/BUILD to core, it causes a circular dependency.
```
pokutuna@instance-1:~/dataform$ bazel test //tests/sqlx/...
ERROR: /home/pokutuna/dataform/sqlx/BUILD:5:11: in ts_library rule //sqlx:sqlx: cycle in dependency graph:
    //tests/sqlx:format.spec (9bf3c46410944eee39af2a199012f4cb7dbe9d88374c707ffb94070f4c0045b9)
    //tests/sqlx:tests (15696f0530617362d5668c59388c0b9089022219dd20111c87cd2937b947b8fe)
.-> //sqlx:sqlx (15696f0530617362d5668c59388c0b9089022219dd20111c87cd2937b947b8fe)
|   //core:core (15696f0530617362d5668c59388c0b9089022219dd20111c87cd2937b947b8fe)
`-- //sqlx:sqlx (15696f0530617362d5668c59388c0b9089022219dd20111c87cd2937b947b8fe)
ERROR: Analysis of target '//tests/sqlx:format.spec' failed; build aborted
```

How should this be resolved?
If I cut out them to `//core/adapters` as a new build target, the dependency will not cycle.
- core -> sqlx
- core -> core/adapters
- sqlx -> core/adapters

Or it could move only some definitions to common/warehouses.
(I'm not familiar with bazel and I'm speaking from a feeling )

I feel these ideas make more changes, so if I need to create a new BUILD, I will do it in another separate PR. After that, I rebase this.
Can you give me some direction on how to fix it?

Reply:
I think the issue is that the `//sqlx:sqlx` BUILD rule is too large, i.e. it's pulling in both lexing/parsing AND formatting.

I'd suggest cutting `//sqlx:sqlx` into two rules: `format` and `sqlx`. `format` will depend on `sqlx`, and only include `format.ts` in its `srcs`. `sqlx` will include all other files (you can use `exclude` to do this: https://bazel.build/reference/be/functions#glob).

Then we'll have:
`//sqlx:format` -> `//sqlx:sqlx`
`//sqlx:format` -> `//core:core`
`//core:core` -> `//sqlx:sqlx`

i.e. no cycles.

Reply:
Thanks for the suggestion, I made another PR.
#1499

Reply:
Great, thanks. Merge that in here and update this PR and I will approve!

(Also, if you increment the minor version in `version.bzl`, I will publish a new CLI version with your changes.)